### PR TITLE
OSError tweaks

### PR DESCRIPTION
**Pull request by crusaderky**
Follow-up to #4741
Related to #4989

### CODE REVIEW

File: distributed/core.py [hunk 1]
```diff
@@ -579,7 +579,7 @@ async def handle_stream(self, comm, extra=None, every_cycle=[]):
                     else:
                         func()
 
-        except (CommClosedError, OSError):
+        except OSError:
```
Comment (crusaderky) [inline distributed/core.py:582]: CommClosedError is a subclass of OSError

File: distributed/nanny.py [hunk 2]
```diff
@@ -279,7 +279,7 @@ async def start(self):
                 if len(ports) > 1 and e.errno == errno.EADDRINUSE:
                     continue
                 else:
-                    raise e
+                    raise
```
Comment (crusaderky) [inline distributed/nanny.py:282]: This is subtly different, as it retains the original traceback!

**Review by fjetter — APPROVED** [review]
LGTM

**Comment by crusaderky** [comment]
@fjetter @jrbourbeau ready for merge

**Pull request merged**